pallets/interest-accrual/src/lib.rs

```diff
@@ -490,18 +473,21 @@ pub mod pallet {
 		}
 
 		pub fn get_rate(
-			interest_rate_per_sec: T::InterestRate,
+			interest_rate_per_year: T::InterestRate,
 		) -> Result<RateDetailsOf<T>, DispatchError> {
+			let interest_rate_per_sec = unchecked_conversion(interest_rate_per_year)?;
 			Rates::<T>::get()
 				.into_iter()
 				.find(|rate| rate.interest_rate_per_sec == interest_rate_per_sec)
 				.ok_or_else(|| Error::<T>::NoSuchRate.into())
 		}
 
-		pub(crate) fn validate_rate(interest_rate_per_year: T::InterestRate) -> DispatchResult {
+		pub(crate) fn validate_interest_rate(
+			interest_rate_per_year: T::InterestRate,
+		) -> DispatchResult {
 			let four_decimals = T::InterestRate::saturating_from_integer(10000);
 			ensure!(
-				interest_rate_per_year < One::one()
+				interest_rate_per_year <= T::InterestRate::saturating_from_integer(2)
```

Comment:
This would require another issue to tackle correctly. This `2` comes because is the "maximum" reasonable value coming from `pallet-loans`, which is a possible interest rate `< 1` plus a penalty from `[0, 1]`. Which in the worst cases could be near to `2`.

Theoretically, there is no upper bound for this, nevertheless, technically, the correct value that should be here is the maximum rate value that does not cause overflow after the whole life of centrifuge change accruing that rate (1000 years for example?).

Reply:
Maybe add a constant `MAX_INTEREST_RATE = 2`, and add
```rust
// Possible interest rates < 1 plus a penalty from [0, 1]
// Which in the worst cases could be near to 2.
```
as a comment to this constant?

Reply:
Agree!